controllers/database_reconciler_helpers.go

```diff
@@ -95,22 +99,28 @@ func (r *DatabaseReconciler) handleDelete(ctx context.Context, database *ndbv1al
 		log.Info("Checking if database instance exists")
 		allDatabases, err := ndb_api.GetAllDatabases(ctx, ndbClient)
 		if err != nil {
-			log.Error(err, "An error occurred while trying to get all databases")
+			errStatement := "An error occurred while trying to get all databases"
```

Comment:
while you are making a change to this line, can you also pls improve the error msg:
"Error fetching NDB database list"